script.js

```diff
@@ -1,4 +1,97 @@
+// intialise mode in the begining
+var mode = 'player 1 mode';
+
+// create two arrays for each dice
+var player1sChanged = [];
+var player2sChanged = [];
+// this 'changed' name for this variable just denotes that i am gonna change this variable
+// create an array for the dice concat numbers
+var player1KeptDaSame = [];
+var player2KeptDaSame = [];
+
+// both players names are blank, i will clone the 'changed' array using the concat fucntion in line 32, this denotes the unchanged rolls.
+// while the 'changed' variable(that is in the name in lines 5 & 6) contains the changed rolls should they want
+
+// generate random dice roll will use this singular function to roll the dice
+var randomDiceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input == '') {
+    mode = 'rolling dice mode for player 1';
+    console.log(mode);
+    // if in player 1 mode output randomly generated dice roll
+    player1sChanged.push(randomDiceRoll(), randomDiceRoll());
+
+    mode = 'dice concat mode for player 1';
+    console.log(mode);
+    console.log('type ok after you say yes or no');
+    // user interface
+    return 'player 1 your rolls are ' + player1sChanged + ' do you want to change the configuration of your numbers?';
+  }
+
+  var player1KeptDaSame = player1sChanged.concat(player1KeptDaSame);
+
+  if (input == 'no' && mode == 'dice concat mode for player 1') {
+    return 'your dice arrangement is the same, at' + player1KeptDaSame;
+  }
+  if (input == 'yes' && mode == 'dice concat mode for player 1') {
+    var temp = this.player1sChanged[0];
+    this.player1sChanged[0] = this.player1sChanged[1];
```

Comment:
Does this code work? What is `this.player1sChanged` and how is it different from `player1sChanged`?